techmatters/terraso-mobile-client
# feat: info sheets component

## Description

Reimplement `InfoSheet` to design specification and standardize usages. Also simplify `InfoButton` usage in application, factoring out `InfoSheetButton` and improving clarity of API (all usages pass in a `sheetHeading` component which is typically `Heading` or `TranslatedHeading` rather than raw strings to `Header` property.) Organize info/help/etc components together to improve cleanliness of `components` directory.

Remaining usages of `OverlaySheet` are all for input/form use-cases, not info use-cases, which will be handled in the Forms component work (https://github.com/techmatters/terraso-mobile-client/issues/1774).

### Checklist

### Related Issues
Fixes #697

### Verification steps

Verify that all info sheets (LandPKS, privacy, soil ID, etc) open to consistent height, have the `BigCloseButton` which functions as expected, and are appropriately styled.
